src/System.Private.CoreLib/shared/System/Runtime/CompilerServices/YieldAwaitable.cs

```diff
@@ -188,7 +187,7 @@ private static Action OutputCorrelationEtwEvent(Action continuation)
             private static readonly SendOrPostCallback s_sendOrPostCallbackRunAction = RunAction;
             /// <summary>Runs an Action delegate provided as state.</summary>
             /// <param name="state">The Action delegate to invoke.</param>
-            private static void RunAction(object state) { ((Action)state)(); }
+            private static void RunAction(object? state) { ((Action)state!)(); }
```

Comment:
Why accept null here and then use `!`?

Reply:
`RunAction` is being used with a delegate that takes an `Action<object?>` because that delegate is passed to a method that expects an `Action<object?>`.

Reply:
Thanks for elaborating.